codes/classical/bits/batch.yml

```diff
@@ -0,0 +1,43 @@
+# UTF-8 unicode encoding
+
+code_id: batch
+
+physical: bits
+
+logical: bits
+
+name: 'Batch code \cite{doi:10.1145/1007352.1007396}'
+
+description: 'Batch codes are a family of codes designed for minimizing the total amount of storage and the worst-case maximal load on any devices in a distributed system. A \((n,N,k,m,t)\) batch code encodes a length-\(n\) string \(x_1,\cdots,x_n\) into an \(m\)-tuple of strings of total length \(N\) (they are also called buckets), such that for each \(k\)-tuple of distinct indices \(i_1,i_2,...,i_k\), the entries \(x_{i_1},...,x_{i_k}\) can be decoded by reading at most \(t\)-symbols from each bucket. Typically, \(n,k\) are given parameters and people try to minimize \(N,m\) as a function of these given parameters. If a each bucket of a batch code contains a single symbol, then we call such \((n,N,k,m)\) batch code is primitive. A multi-user generalization of batch code is named multiset batch code. If any multiset \(i_1, i_2, ..., i_k \in [n]\) there is a partition of buckets into subset \(S_1, ..., S_k \subset [m]\) such that each \(x_{i_j}\) can be recovered by reading at most one symbol from each bucket in \(S_j\), then we call such a \((n,N,k,m,1)\) batch code to a \((n, N, k, m)\) multiset batch code. Combining two batch codes \(C_1\) and \(C_2\) which are \((n_1,N_1,k_1,m_1)\) and \((n_2,N_2,k_2,m_2)\) batch codes respectively, we can obatin a composed batch code \((C_1\otimes C_2)\) which is a \((n_1, m_1N_2, k_1 k_2, m_1 m_2)\) batch code.'
+
+#decoder: 'Minimum-weight \href{https://some.com/page}{perfect} matching \cite{ABCD_YYYY_journalorconf_xyz}'
+
+#decoder: 'Here''s an example'
+
+#features:
+
+ 
+
+  
+
+relations:
+
+  parents:
+
+    #
+    #
+    # - code_id: 'stabilizer'
+    #   detail: 'fsfdjknasjfkldsn'
+
+    - code_id: 'expander'
+      detail: 'Instead of simply replicating data into multiple devices, one can distribute items to buckets based on an unbalanced expander graph. Using random graphs of polynomial size, we can achieve \(N/n=O(\log n)\) and \(m=O(k)\).'
+
+    - code_id: 'smooth_code'
+      detail: 'If we have a \(q\)-query smooth code \(C:\Sigma^n \rightarrow \Sigma^m\). Then \(C\) is a primitive multiset batch code with the same parameters as the smooth code, where \(\lfloor k \rfloor\).'
+    
+    - code_id: 'reed_muller'
```

Comment:
Expander codes have little relation to batch codes; discard. It is true that the graph can be an expander, but this does not mean that expander codes have much in common with them.
It is also not clear how Reed-Muller codes are related. Smooth code is not an established term.
The true relatives are locally recoverable codes with availability.

Reply:
parent is then locally_recoverable (see https://github.com/errorcorrectionzoo/eczoo_data/pull/135/files), which itself is a child of codes_with_availability

Reply:
Add another parent code_id: bits_into_bits (since this isn't a linear code)